lib/apivore/validator.rb

```diff
@@ -21,8 +22,7 @@ def matches?(swagger_checker)
         send(
           method,
           full_path(swagger_checker),
-          params['_data'] || {},
-          params['_headers'] || {}
+          *RailsShim.request_params(params)
```

Comment:
Thought this through again, and I think there's a third option.  Ultimately the goal here is to `send` a method with arguments, e.g.

```ruby
# Rails 5
get('/posts', params: {}, headers: {})

# Rails 4
get('/posts', {}, {})
```

The first argument, `method`, isn't really an argument, but a varying method name.  The rest of the arguments (`path`, `params`, `header`) are the arguments passed to that method.

The Rails shim doesn't need to know about the fact that we're using metaprogramming to send an arbitrary request method.  It should just be responsible for taking all the information about the request (`path`, `params`, `header`) and formulating them in a way that is satisfactory to the currently running version of Rails.

So I'd suggest this:

```ruby
request_args = RailsShim.action_dispatch_request_args(
  full_path(swagger_checker),
  params: params['_data'] || {} , 
  headers: params['_header'] || {}
)
send(method, *request_args)
```

Does that seem reasonable?